src/api/app/jobs/staging_project_copy_job.rb

```diff
@@ -0,0 +1,7 @@
+class StagingProjectCopyJob < ApplicationJob
+  def perform(staging_workflow_project_name, original_staging_project_name, staging_project_copy_name)
+    staging_workflow_project = Project.find_by!(name: staging_workflow_project_name)
+    original_staging_project = staging_workflow_project.staging.staging_projects.find_by!(name: original_staging_project_name)
+    original_staging_project.copy(staging_project_copy_name)
```

Comment:
FYI: If this fails, the user has no way to know what happened.

Is there something we can do about this?

Reply:
I'll have a look into that

Reply:
I just pushed a commit for a possible solution. It's not working yet as explained in the commit, but we can discuss about this.

Reply:
I removed the commit from the PR. I will create another PR when this one is merged.